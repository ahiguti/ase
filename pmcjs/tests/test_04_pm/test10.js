
var filename = "dbmfilter.out";

use('Perl');
Perl.Use('GDBM_File');
var wrcr = Perl.Constants['GDBM_File'].GDBM_WRCREAT;
var db = Perl.NewHash();
var tx = Perl.Tie(db, 'GDBM_File', filename, wrcr, 0640);

var filter_store = function(_) { print("STORE[" + _ + "]"); return 'X' + _; }
tx.filter_store_value(filter_store);

db['foo'] = 'bar';
for (var i in db) {
  print("db[" + i + "] = " + db[i]);
}

