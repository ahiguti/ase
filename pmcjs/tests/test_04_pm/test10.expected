STORE[bar]
db[foo] = Xbar
